costools/add_cos_s_region.py

```diff
@@ -0,0 +1,107 @@
+#! /usr/bin/env python
+
+# Add the S_REGION keyword to raw files
+#
+import glob
+import argparse
+
+from calcos import cosutil
+from astropy.io import fits
+
+__version__ = '0.1'
+def main(dry_run=False):
+    """Add the S-region keyword to all COS FITS files in the current
+    directory that meet the following criteria:
+
+    Filename ends with one of ['rawtag.fits',
+                               'rawtag_a.fits',
+                               'rawtag_b.fits',
+                               'rawaccum.fits',
+                               'rawaccum_a.fits',
+                               'rawaccum_b.fits',
+                               'rawacq.fits']
+
+    The S_REGION keyword will have the value
+    'CIRCLE ra_aper dec_aper diameter'
+    where ra_aper and dec_aper are in decimal degrees and are read from
+    the extension 1 header, and diameter = 2.5/3600.0, which is the diameter
+    of the PSA and BOA in decimal degrees.
+
+    The keyword will be added after the PA_APER keyword in the extension 1
+    header
+
+    """
+    endings = ['rawtag.fits',
+               'rawtag_a.fits',
+               'rawtag_b.fits',
+               'rawaccum.fits',
+               'rawaccum_a.fits',
+               'rawaccum_b.fits',
+               'rawacq.fits']
+
+    fitsfiles = glob.glob('*.fits')
+    for file in fitsfiles:
+        for ending in endings:
+            if file.endswith(ending):
+                add_s_region(file, dry_run)
+    return
+
+def add_s_region(file, dry_run):
+
+    open_mode = 'readonly'
+    if not dry_run: open_mode = 'update'
+
+    with fits.open(file, mode=open_mode) as f1:
+        if (len(f1) < 2):
+            cosutil.printWarning("File {} has only {} extensions".format(file, len(f1)))
+            cosutil.printContinuation("Need at least 2")
+            return
+        prihdr = f1[0].header
+        if prihdr['INSTRUME'] != 'COS':
+            cosutil.printWarning("File {}: Instrument is not COS".format(file))
+            return
+
+        ext1hdr = f1[1].header
+        try:
+            ra_aper = ext1hdr['RA_APER']
+            dec_aper = ext1hdr['DEC_APER']
+            pa_aper = ext1hdr['PA_APER']
+        except KeyError:
+            cosutil.printWarning('One or more of RA_APER, DEC_APER, PA_APER is not present')
+            cosutil.printContinuation('in extension 1 header of file {}'.format(file))
+            return
+        diameter = 2.5 / 3600.0
+        s_region = 'CIRCLE {0:.8f} {1:.7f} {2:.8f}'.format(ra_aper, dec_aper, diameter)
```

Comment:
I believe the convention is also to include the coordinate frame btu you sould check with the archive what they expect. So something like 
```
'CIRCLE ICRS {0:.8f} {1:.7f} {2:.8f}'.format(ra_aper, dec_aper, diameter)
```

Reply:
Will insert the ICRS.  STIS version already has it